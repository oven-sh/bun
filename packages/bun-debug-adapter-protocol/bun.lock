{
  "lockfileVersion": 0,
  "workspaces": {
    "": {
      "name": "bun-debug-adapter-protocol",
      "dependencies": {
        "semver": "^7.5.4",
        "source-map-js": "^1.0.2",
      },
    },
  },
  "packages": {
    "lru-cache": ["lru-cache@6.0.0", "", { "dependencies": { "yallist": "^4.0.0" } }, "sha512-Jo6dJ04CmSjuznwJSS3pUeWmd/H0ffTlkXXgwZi+eq1UCmqQwCh+eLsYOYCwY991i2Fah4h1BEMCx4qThGbsiA=="],

    "semver": ["semver@7.5.4", "", { "dependencies": { "lru-cache": "^6.0.0" }, "bin": { "semver": "bin/semver.js" } }, "sha512-1bCSESV6Pv+i21Hvpxp3Dx+pSD8lIPt8uVjRrxAUt/nbswYc+tK6Y2btiULjd4+fnq15PX+nqQDC7Oft7WkwcA=="],

    "source-map-js": ["source-map-js@1.0.2", "", {}, "sha512-R0XvVJ9WusLiqTCEiGCmICCMplcCkIwwR11mOSD9CR5u+IXYdiseeEuXCVAjS54zqwkLcPNnmU4OeJ6tUrWhDw=="],

    "yallist": ["yallist@4.0.0", "", {}, "sha512-3wdGidZyq5PB084XLES5TpOSRA3wjXAlIWMhum2kRcv/41Sn2emQ0dycQW4uZXLejwKvg6EsvbdlVL+FYEct7A=="],
  }
}
