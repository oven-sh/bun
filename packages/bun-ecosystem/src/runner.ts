import type { Package } from "./packages";
import { packages } from "./packages";
import { existsSync, copyFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { globby } from "globby";

for (const pkg of packages) {
  try {
    await loadPackage(pkg, "tmp");
  } catch (error) {
    console.error(pkg.name, error);
  }
}

async function loadPackage(pkg: Package, cwd?: string): Promise<void> {
  await gitClone({
    cwd,
    repository: pkg.repository,
    name: pkg.name,
  });
  const dir = join(cwd ?? "", pkg.name, pkg.cwd ?? "");
  await spawn({
    cwd: dir,
    cmd: ["bun", "install"],
  });
  if (!pkg.tests || pkg.tests.style !== "jest") {
    return;
  }
  const files = await globby(pkg.tests.include, {
    cwd: dir,
    ignore: pkg.tests.exclude ?? [crypto.randomUUID()],
    onlyFiles: true,
    caseSensitiveMatch: false,
  });
  if (!files.length) {
    throw new Error("No tests found");
  }
  for (const file of files) {
    let path = file;
    if (!file.includes(".test.")) {
      const ext = path.lastIndexOf(".");
      path = file.substring(0, ext) + ".test" + file.substring(ext);
      copyFileSync(join(dir, file), join(dir, path));
    }
    await spawn({
      cwd: dir,
      cmd: ["bun", "test", path],
    });
  }
}

type GitCloneOptions = {
  repository: string;
  cwd?: string;
  name?: string;
};

async function gitClone(options: GitCloneOptions): Promise<void> {
  const name = options.name ?? dirname(options.repository);
  const cwd = options.cwd ?? process.cwd();
  const path = join(cwd, name);
  if (existsSync(path)) {
    await spawn({
      cwd: path,
      cmd: ["git", "pull"],
    });
  } else {
    const url = `${options.repository}`;
    await spawn({
      cwd,
      cmd: ["git", "clone", "--single-branch", "--depth", "1", url, name],
    });
  }
}

type SpawnOptions = {
  cwd: string;
  cmd: string[];
};

async function spawn({ cwd, cmd }: SpawnOptions) {
  const { exited } = await Bun.spawn({
    cwd,
    cmd,
    stdout: "inherit",
    stderr: "inherit",
  });
  const exitCode = await exited;
  if (exitCode !== 0) {
    throw new Error(`"${cmd.join(" ")}" exited with ${exitCode}`);
  }
}
