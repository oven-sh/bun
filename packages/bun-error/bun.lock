{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "bun-error",
      "dependencies": {
        "preact": "^10.27.2",
      },
    },
  },
  "packages": {
    "preact": ["preact@10.27.2", "", {}, "sha512-5SYSgFKSyhCbk6SrXyMpqjb5+MQBgfvEKE/OC+PujcY34sOpqtr+0AZQtPYx5IA6VxynQ7rUPCtKzyovpj9Bpg=="],
  }
}
