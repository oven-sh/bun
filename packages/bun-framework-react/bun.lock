{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "bun-framework-react",
      "dependencies": {
        "react": "0.0.0-experimental-a757cb76-20251002",
        "react-dom": "0.0.0-experimental-a757cb76-20251002",
        "react-refresh": "0.0.0-experimental-a757cb76-20251002",
      },
      "devDependencies": {
        "@types/react": "^19.1.13",
        "@types/react-dom": "^19.1.9",
      },
    },
  },
  "packages": {
    "@types/react": ["@types/react@19.1.13", "", { "dependencies": { "csstype": "^3.0.2" } }, "sha512-hHkbU/eoO3EG5/MZkuFSKmYqPbSVk5byPFa3e7y/8TybHiLMACgI8seVYlicwk7H5K/rI2px9xrQp/C+AUDTiQ=="],

    "@types/react-dom": ["@types/react-dom@19.1.9", "", { "peerDependencies": { "@types/react": "^19.0.0" } }, "sha512-qXRuZaOsAdXKFyOhRBg6Lqqc0yay13vN7KrIg4L7N4aaHN68ma9OK3NE1BoDFgFOTfM7zg+3/8+2n8rLUH3OKQ=="],

    "csstype": ["csstype@3.1.3", "", {}, "sha512-M1uQkMl8rQK/szD0LNhtqxIPLpimGm8sOBwU7lLnCpSbTyY3yeU1Vc7l4KT5zT4s/yOxHH5O7tIuuLOCnLADRw=="],

    "react": ["react@0.0.0-experimental-a757cb76-20251002", "", {}, "sha512-7ZcE4sSUGgrXgUWa84iwC9DqwDFbQBgffFmu2DoNqFseruA/JjxQDXKwpV5acdxOM/0uzfSGrapHU3C3ZLiU2g=="],

    "react-dom": ["react-dom@0.0.0-experimental-a757cb76-20251002", "", { "dependencies": { "scheduler": "0.0.0-experimental-a757cb76-20251002" }, "peerDependencies": { "react": "0.0.0-experimental-a757cb76-20251002" } }, "sha512-XjIkmW8mMx9kURHJUY+dhv1Ugan3RmEJIwrZEbAFcJA4S8RXL1wl+xsQJpDCh8kmeX/n25VAmFY8/j1MzqUHfA=="],

    "react-refresh": ["react-refresh@0.0.0-experimental-a757cb76-20251002", "", {}, "sha512-uYd+N2W8/LymZQyY5u1BMWVvLlBV+5SxztBsFjOGuitE4x7sSCj8TwgS+8bxIEBucEVJglfOhDPCPohL/uEQdg=="],

    "scheduler": ["scheduler@0.0.0-experimental-a757cb76-20251002", "", {}, "sha512-YCVGuzmF7u5HIpOdPFD4tZTPzQlOrtViag7uaWjJXfFx37C8sypfNeSXNXoYJeT/ICybxP1EsbHh2oByQHC2Cg=="],
  }
}
