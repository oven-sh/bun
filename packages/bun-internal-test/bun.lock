{
  "lockfileVersion": 0,
  "workspaces": {
    "": {
      "dependencies": {
        "@actions/core": "latest",
        "p-queue": "^8.0.1",
      },
      "devDependencies": {
        "@types/p-queue": "^3.2.1",
        "bun-types": "canary",
        "prettier": "^2.8.2",
      },
    },
    "runners/bun": {
      "name": "bun",
    },
    "runners/qunit": {
      "name": "qunit",
    },
  },
  "packages": {
    "@actions/core": ["@actions/core@1.10.1", "", { "dependencies": { "@actions/http-client": "^2.0.1", "uuid": "^8.3.2" } }, "sha512-3lBR9EDAY+iYIpTnTIXmWcNbX3T2kCkAEQGIQx4NVQ0575nk2k3GRZDTPQG+vVtS2izSLmINlxXf0uLtnrTP+g=="],

    "@actions/http-client": ["@actions/http-client@2.1.1", "", { "dependencies": { "tunnel": "^0.0.6" } }, "sha512-qhrkRMB40bbbLo7gF+0vu+X+UawOvQQqNAA/5Unx774RS8poaOhThDOG6BGmxvAnxhQnDp2BG/ZUm65xZILTpw=="],

    "@types/p-queue": ["@types/p-queue@3.2.1", "", { "dependencies": { "p-queue": "*" } }, "sha512-tgAdn5zEs05NuHzOyRM34cMO0rczStphR/kLo/ZJwwwJ5S2+QVxwA6gST3vDHWPB1oDfUuT6wOouhJvJkBCA0w=="],

    "bun": ["bun@workspace:runners/bun", {}],

    "bun-types": ["bun-types@1.0.4-canary.20230929T233451", "", {}, "sha512-Ke/y3GX0T2ZYKx0UKKCnFRRP9bXzVRcRZCrIlF5/aRNEpgbfrMBw+s5imPqrV2CqJ7q6kIbMikCahzMr4N9PTg=="],

    "eventemitter3": ["eventemitter3@5.0.1", "", {}, "sha512-GWkBvjiSZK87ELrYOSESUYeVIc9mvLLf/nXalMOS5dYrgZq9o5OVkbZAVM06CVxYsCwH9BDZFPlQTlPA1j4ahA=="],

    "p-queue": ["p-queue@8.0.1", "", { "dependencies": { "eventemitter3": "^5.0.1", "p-timeout": "^6.1.2" } }, "sha512-NXzu9aQJTAzbBqOt2hwsR63ea7yvxJc0PwN/zobNAudYfb1B7R08SzB4TsLeSbUCuG467NhnoT0oO6w1qRO+BA=="],

    "p-timeout": ["p-timeout@6.1.2", "", {}, "sha512-UbD77BuZ9Bc9aABo74gfXhNvzC9Tx7SxtHSh1fxvx3jTLLYvmVhiQZZrJzqqU0jKbN32kb5VOKiLEQI/3bIjgQ=="],

    "prettier": ["prettier@2.8.8", "", { "bin": { "prettier": "bin-prettier.js" } }, "sha512-tdN8qQGvNjw4CHbY+XXk0JgCXn9QiF21a55rBe5LJAU+kDyC4WQn4+awm2Xfk2lQMk5fKup9XgzTZtGkjBdP9Q=="],

    "qunit": ["qunit@workspace:runners/qunit", {}],

    "tunnel": ["tunnel@0.0.6", "", {}, "sha512-1h/Lnq9yajKY2PEbBadPXj3VxsDDu844OnaAo52UVmIzIvwwtBPIuNvkjuzBlTWpfJyUbG3ez0KSBibQkj4ojg=="],

    "uuid": ["uuid@8.3.2", "", { "bin": { "uuid": "dist/bin/uuid" } }, "sha512-+NYs2QeMWy+GWFOEm9xnn6HCDp0l7QBD7ml8zLUmJ+93Q5NF0NocErnwkTkXVFNiX3/fpC6afS8Dhb/gz7R7eg=="],
  }
}
