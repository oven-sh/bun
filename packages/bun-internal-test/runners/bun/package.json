{
  "private": true,
  "name": "bun",
  "module": "runner.ts"
}
