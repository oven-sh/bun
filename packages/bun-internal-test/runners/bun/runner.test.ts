import { describe, test, expect } from "bun:test";
import { tmpdir } from "node:os";
import { mkdtempSync, mkdirSync, writeFileSync } from "node:fs";
import { join } from "node:path";
import type { FindTestOptions, ParseTestResult, RunTestResult } from "./runner";
import { bunSpawn, nodeSpawn, findTests, parseTest, runTest, runTests } from "./runner";

describe("runTests()", () => {
  const cwd = createFs({
    "example1.test.ts": "",
    "example2.spec.js": `
      import { test, expect } from "bun:test";

      test("this should pass", () => {
        expect(true).toBe(true);
      });
    `,
    "example3.test.mjs": `
      import { test, expect } from "bun:test";

      test("this should fail", () => {
        expect(true).toBe(false);
      });
    `,
    "path": {
      "to": {
        "example4.test.ts": `
          import { test, expect } from "bun:test";

          test.skip("this should skip", () => {
            expect(true).toBe(true);
          });
        `
      }
    }
  });
  test("can run all tests", async () => {
    const results = runTests({ cwd });
    while (true) {
      const { value, done } = await results.next();
      toMatchResult(value);
      if (done) {
        break;
      }
    }
  });
});

describe("runTest()", () => {
  const cwd = createFs({
    "example1.ts": `
      import { test, expect } from "bun:test";

      test("this should pass", () => {
        expect(true).toBe(true);
      });

      test("this should fail", () => {
        expect(true).toBe(false);
      });

      test.skip("this should skip", () => {
        expect(true).toBe(true);
      });
    `,
    "path": {
      "to": {
        "example2.test.ts": `
          import { test, expect } from "bun:test";

          test("this should pass", () => {
            expect(true).toBe(true);
          });
        `
      }
    },
    "preload": {
      "preload.test.ts": `
        import { test, expect } from "bun:test";

        test("test should have preloaded", () => {
          expect(globalThis.preload).toBe(true);
        });
      `,
      "preload.ts": `
        globalThis.preload = true;
      `
    }
  });
  test("can run a test", async () => {
    const result = await runTest({
      cwd,
      path: "path/to/example2.ts",
    });
    toMatchResult(result);
  });
  test("can run a test with a symlink", async () => {
    const result = await runTest({
      cwd,
      path: "example1.ts",
    });
    toMatchResult(result);
  });
  test("can run a test with a preload", async () => {
    const result = await runTest({
      cwd,
      path: "preload/preload.test.ts",
      preload: ["./preload/preload.ts"]
    });
    toMatchResult(result);
  });
});

describe("parseTest()", () => {
  const cwd = createFs({
    "example1.test.ts": `
      import { test, expect } from "bun:test";

      test("this should pass", () => {
        expect(true).toBe(true);
      });

      test("this should fail", () => {
        expect(true).toBe(false);
      });

      test.skip("this should skip", () => {
        expect(true).toBe(true);
      });
    `,
    "path": {
      "to": {
        "example2.test.js": "",
        "example3.spec.tsx": `
          import { describe, test, expect } from "bun:test";

          describe("tests", () => {
            test("this should pass", () => {
              expect(true).toBe(true);
            });

            test("this should fail", () => {
              throw new TypeError("Oops!");
            });
          });
        `,
      }
    }
  });
  test("can parse test results", async () => {
    const { stderr } = await bunSpawn({
      cwd,
      cmd: "bun",
      args: ["test"],
    });
    const result = parseTest(stderr, { cwd });
    toMatchResult(result);
  });
});

function toMatchResult(result: ParseTestResult | RunTestResult): void {
  result.summary.duration = 0;
  result.info.revision = "";
  result.info.version = "";
  result.info.os = undefined;
  result.info.arch = undefined;
  for (const file of result.files) {
    file.summary.duration = 0;
  }
  if ("stderr" in result) {
    result.stderr = "";
    result.stdout = "";
  }
  expect(result).toMatchSnapshot();
}

describe("findTests()", () => {
  const cwd = createFs({
    "readme.md": "",
    "package.json": "",
    "path": {
      "to": {
        "example1.js": "",
        "example2.test.ts": "",
        "example3.spec.js": "",
        "example.txt": "",
      },
      "example4.js.map": "",
      "example4.js": "",
      "example5.test.ts": "",
    }
  });
  const find = (options: FindTestOptions = {}) => {
    const results = findTests({ cwd, ...options });
    return [...results].sort();
  };
  test("can find all tests", () => {
    const results = find();
    expect(results).toEqual([
      "path/example4.js",
      "path/example5.test.ts",
      "path/to/example1.js",
      "path/to/example2.test.ts",
      "path/to/example3.spec.js",
    ]);
  });
  test("can find tests that match a directory", () => {
    const results = find({
      filters: ["path/to/"],
    });
    expect(results).toEqual([
      "path/to/example1.js",
      "path/to/example2.test.ts",
      "path/to/example3.spec.js",
    ]);
  });
  test("can find tests that match a file", () => {
    const results = find({
      filters: [
        "example1.js",
        "example5.test.ts"
      ],
    });
    expect(results).toEqual([
      "path/example5.test.ts",
      "path/to/example1.js",
    ]);
  });
  test("can find tests that match a glob", () => {
    const results = find({
      filters: [
        "path/to/*.js",
        "*.spec.*",
      ],
    });
    expect(results).toEqual([
      "path/to/example1.js",
      "path/to/example3.spec.js",
    ]);
  });
  test("can find no tests", () => {
    const results = find({
      filters: [
        "path/to/nowhere/*",
      ],
    });
    expect(results).toEqual([]);
  });
});

describe("bunSpawn()", () => {
  testSpawn(bunSpawn);
});

describe("nodeSpawn()", () => {
  testSpawn(nodeSpawn);
});

function testSpawn(spawn: typeof bunSpawn): void {
  test("can run a command", async () => {
    const { exitCode, stdout, stderr } = await spawn({
      cmd: "echo",
      args: ["hello world"],
    });
    expect(exitCode).toBe(0);
    expect(stdout).toBe("hello world\n");
    expect(stderr).toBe("");
  });
  test("can timeout a command", async () => {
    const { exitCode, stdout, stderr } = await spawn({
      cmd: "sleep",
      args: ["60"],
      timeout: 1,
    });
    expect(exitCode).toBe(null);
    expect(stdout).toBe("");
    expect(stderr).toBe("");
  });
}

type FsTree = {
  [path: string]: FsTree | string;
};

function createFs(tree: FsTree): string {
  let cwd = mkdtempSync(join(tmpdir(), "bun-internal-test-"));
  if (cwd.startsWith("/var/folders")) {
    cwd = join("/private", cwd); // HACK: macOS
  }
  const traverse = (tree: FsTree, path: string) => {
    for (const [name, content] of Object.entries(tree)) {
      const newPath = join(path, name);
      if (typeof content === "string") {
        writeFileSync(newPath, content);
      } else {
        mkdirSync(newPath);
        traverse(content, newPath);
      }
    }
  };
  traverse(tree, cwd);
  return cwd;
}
