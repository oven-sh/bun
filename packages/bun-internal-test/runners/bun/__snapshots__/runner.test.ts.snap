// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`runTests() can run all tests 1`] = `
{
  "exitCode": 1,
  "files": [
    {
      "file": "path/to/example4.test.ts",
      "status": "fail",
      "summary": {
        "duration": 1,
        "fail": 1,
        "files": 1,
        "pass": 0,
        "skip": 1,
        "tests": 1,
        "todo": 2,
      },
      "tests": [
        {
          "duration": 0,
          "name": "this should skip",
          "status": "skip",
        },
        {
          "duration": 0,
          "name": "this should todo",
          "status": "todo",
        },
        {
          "duration": 0,
          "errors": [
            {
              "message": "expect(received).toBe(expected)\n\nExpected: false\nReceived: true\n",
              "name": "Error",
              "preview": "10 |           test.todo(\"this should todo and fail\", () => {\n11 |             expect(true).toBe(false);\n                ^",
              "stack": [
                {
                  "column": 12,
                  "file": "path/to/example4.test.ts",
                  "function": undefined,
                  "line": 11,
                },
              ],
            },
          ],
          "name": "this should todo and fail",
          "status": "todo",
        },
        {
          "duration": 0,
          "name": "this should todo and pass",
          "status": "fail",
        },
      ],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "stderr": "",
  "stdout": "",
  "summary": {
    "duration": 1,
    "fail": 1,
    "files": 1,
    "pass": 0,
    "skip": 1,
    "tests": 1,
    "todo": 2,
  },
}
`;

exports[`runTests() can run all tests 2`] = `
{
  "exitCode": 0,
  "files": [
    {
      "file": "example2.spec.js",
      "status": "pass",
      "summary": {
        "duration": 1,
        "fail": 0,
        "files": 1,
        "pass": 1,
        "skip": 0,
        "tests": 1,
        "todo": 0,
      },
      "tests": [
        {
          "duration": 1,
          "name": "this should pass",
          "status": "pass",
        },
      ],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "stderr": "",
  "stdout": "",
  "summary": {
    "duration": 1,
    "fail": 0,
    "files": 1,
    "pass": 1,
    "skip": 0,
    "tests": 1,
    "todo": 0,
  },
}
`;

exports[`runTests() can run all tests 3`] = `
{
  "exitCode": 1,
  "files": [
    {
      "file": "example3.test.mjs",
      "status": "fail",
      "summary": {
        "duration": 1,
        "fail": 2,
        "files": 1,
        "pass": 0,
        "skip": 0,
        "tests": 2,
        "todo": 0,
      },
      "tests": [
        {
          "duration": 1,
          "errors": [
            {
              "message": "expect(received).toBe(expected)\n\nExpected: false\nReceived: true\n",
              "name": "Error",
              "preview": "1 | \n2 |       import { test, expect } from \"bun:test\";\n3 | \n4 |       test(\"this should fail\", () => {\n5 |         expect(true).toBe(false);\n            ^",
              "stack": [
                {
                  "column": 8,
                  "file": "example3.test.mjs",
                  "function": undefined,
                  "line": 5,
                },
              ],
            },
          ],
          "name": "this should fail",
          "status": "fail",
        },
        {
          "duration": 1,
          "errors": [
            {
              "message": "test \"this should timeout\" timed out after 1ms",
              "name": "Timeout",
            },
          ],
          "name": "this should timeout",
          "status": "fail",
        },
      ],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "stderr": "",
  "stdout": "",
  "summary": {
    "duration": 1,
    "fail": 2,
    "files": 1,
    "pass": 0,
    "skip": 0,
    "tests": 2,
    "todo": 0,
  },
}
`;

exports[`runTests() can run all tests 4`] = `
{
  "exitCode": 0,
  "files": [
    {
      "file": "example1.test.ts",
      "status": "pass",
      "summary": {
        "duration": 1,
        "fail": 0,
        "files": 1,
        "pass": 0,
        "skip": 0,
        "tests": 0,
        "todo": 0,
      },
      "tests": [],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "stderr": "",
  "stdout": "",
  "summary": {
    "duration": 1,
    "fail": 0,
    "files": 1,
    "pass": 0,
    "skip": 0,
    "tests": 0,
    "todo": 0,
  },
}
`;

exports[`runTests() can run all tests 5`] = `
{
  "files": [
    {
      "file": "path/to/example4.test.ts",
      "status": "fail",
      "summary": {
        "duration": 1,
        "fail": 1,
        "files": 1,
        "pass": 0,
        "skip": 1,
        "tests": 1,
        "todo": 2,
      },
      "tests": [
        {
          "duration": 0,
          "name": "this should skip",
          "status": "skip",
        },
        {
          "duration": 0,
          "name": "this should todo",
          "status": "todo",
        },
        {
          "duration": 0,
          "errors": [
            {
              "message": "expect(received).toBe(expected)\n\nExpected: false\nReceived: true\n",
              "name": "Error",
              "preview": "10 |           test.todo(\"this should todo and fail\", () => {\n11 |             expect(true).toBe(false);\n                ^",
              "stack": [
                {
                  "column": 12,
                  "file": "path/to/example4.test.ts",
                  "function": undefined,
                  "line": 11,
                },
              ],
            },
          ],
          "name": "this should todo and fail",
          "status": "todo",
        },
        {
          "duration": 0,
          "name": "this should todo and pass",
          "status": "fail",
        },
      ],
    },
    {
      "file": "example2.spec.js",
      "status": "pass",
      "summary": {
        "duration": 1,
        "fail": 0,
        "files": 1,
        "pass": 1,
        "skip": 0,
        "tests": 1,
        "todo": 0,
      },
      "tests": [
        {
          "duration": 1,
          "name": "this should pass",
          "status": "pass",
        },
      ],
    },
    {
      "file": "example3.test.mjs",
      "status": "fail",
      "summary": {
        "duration": 1,
        "fail": 2,
        "files": 1,
        "pass": 0,
        "skip": 0,
        "tests": 2,
        "todo": 0,
      },
      "tests": [
        {
          "duration": 1,
          "errors": [
            {
              "message": "expect(received).toBe(expected)\n\nExpected: false\nReceived: true\n",
              "name": "Error",
              "preview": "1 | \n2 |       import { test, expect } from \"bun:test\";\n3 | \n4 |       test(\"this should fail\", () => {\n5 |         expect(true).toBe(false);\n            ^",
              "stack": [
                {
                  "column": 8,
                  "file": "example3.test.mjs",
                  "function": undefined,
                  "line": 5,
                },
              ],
            },
          ],
          "name": "this should fail",
          "status": "fail",
        },
        {
          "duration": 1,
          "errors": [
            {
              "message": "test \"this should timeout\" timed out after 1ms",
              "name": "Timeout",
            },
          ],
          "name": "this should timeout",
          "status": "fail",
        },
      ],
    },
    {
      "file": "example1.test.ts",
      "status": "pass",
      "summary": {
        "duration": 1,
        "fail": 0,
        "files": 1,
        "pass": 0,
        "skip": 0,
        "tests": 0,
        "todo": 0,
      },
      "tests": [],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "summary": {
    "duration": 1,
    "fail": 3,
    "files": 4,
    "pass": 1,
    "skip": 1,
    "tests": 4,
    "todo": 2,
  },
}
`;

exports[`runTest() can run a test 1`] = `
{
  "exitCode": 0,
  "files": [
    {
      "file": "example2.test.ts",
      "status": "pass",
      "summary": {
        "duration": 1,
        "fail": 0,
        "files": 1,
        "pass": 1,
        "skip": 0,
        "tests": 1,
        "todo": 0,
      },
      "tests": [
        {
          "duration": 1,
          "name": "this should pass",
          "status": "pass",
        },
      ],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "stderr": "",
  "stdout": "",
  "summary": {
    "duration": 1,
    "fail": 0,
    "files": 1,
    "pass": 1,
    "skip": 0,
    "tests": 1,
    "todo": 0,
  },
}
`;

exports[`runTest() can run a test with a symlink 1`] = `
{
  "exitCode": 1,
  "files": [
    {
      "file": "example1.ts",
      "status": "fail",
      "summary": {
        "duration": 1,
        "fail": 1,
        "files": 1,
        "pass": 1,
        "skip": 1,
        "tests": 2,
        "todo": 1,
      },
      "tests": [
        {
          "duration": 1,
          "name": "this should pass",
          "status": "pass",
        },
        {
          "duration": 1,
          "errors": [
            {
              "message": "expect(received).toBe(expected)\n\nExpected: false\nReceived: true\n",
              "name": "Error",
              "preview": "4 |       test(\"this should pass\", () => {\n5 |         expect(true).toBe(true);\n6 |       });\n7 | \n8 |       test(\"this should fail\", () => {\n9 |         expect(true).toBe(false);\n            ^",
              "stack": [
                {
                  "column": 8,
                  "file": "example1.ts",
                  "function": undefined,
                  "line": 9,
                },
              ],
            },
          ],
          "name": "this should fail",
          "status": "fail",
        },
        {
          "duration": 0,
          "name": "this should skip",
          "status": "skip",
        },
        {
          "duration": 0,
          "name": "this should todo",
          "status": "todo",
        },
      ],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "stderr": "",
  "stdout": "",
  "summary": {
    "duration": 1,
    "fail": 1,
    "files": 1,
    "pass": 1,
    "skip": 1,
    "tests": 2,
    "todo": 1,
  },
}
`;

exports[`runTest() can run a test with a preload 1`] = `
{
  "exitCode": 0,
  "files": [
    {
      "file": "preload.test.ts",
      "status": "pass",
      "summary": {
        "duration": 1,
        "fail": 0,
        "files": 1,
        "pass": 1,
        "skip": 0,
        "tests": 1,
        "todo": 0,
      },
      "tests": [
        {
          "duration": 1,
          "name": "test should have preloaded",
          "status": "pass",
        },
      ],
    },
  ],
  "info": {
    "arch": undefined,
    "name": "bun test",
    "os": undefined,
    "revision": "",
    "version": "",
  },
  "stderr": "",
  "stdout": "",
  "summary": {
    "duration": 1,
    "fail": 0,
    "files": 1,
    "pass": 1,
    "skip": 0,
    "tests": 1,
    "todo": 0,
  },
}
`;
